internal/pkg/template/diff/override.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import "gopkg.in/yaml.v3"
+
+type Overrider interface {
+	Match(from, to *yaml.Node, key string) bool
+	Parse(from, to *yaml.Node, key string) (diffNode, error)
```

Comment:
Can we get away with these methods being private? A client wouldn't be able to implement a test double Overrider themselves anyway since `diffNode` isn't exported